system/jlib/jlzw.cpp

```diff
@@ -2224,6 +2248,10 @@ class CCompressedFile : implements ICompressedFileIO, public CInterface
                         compMethod = COMPRESS_METHOD_LZW;
                         expander.setown(createLZWExpander(true));
                     }
+                    //Preallocate the expansion target to the block size - to ensure it is the right size and
+                    //avoid reallocation when expanding lz4
+                    curblockbuf.reserve(trailer.blockSize);
+                    curblockbuf.clear();
```

Comment:
should the abobe 2 lines be ensureCapacity ?

Reply:
blockSize is the compressed block size, but curblockbuf is used to expand into, so it won't be big enough.

Reply:
It will be enough because LZ4 only compresses at most the block size into a single chunk.

Reply:
Yes, that would be better - I couldn't find that function...despite looking.